File: app/src/main/java/cn/alvkeke/dropto/ui/intf/CategoryAttemptListener.java
package cn.alvkeke.dropto.ui.intf;

import cn.alvkeke.dropto.data.Category;

public interface CategoryAttemptListener {
    enum Attempt {
        CREATE,
        REMOVE,
        UPDATE,
        SHOW_DETAIL,
        SHOW_CREATE,
        SHOW_EXPAND,
    }
    void onAttempt(CategoryAttemptListener.Attempt attempt, Category category);
}

File: app/src/main/java/cn/alvkeke/dropto/ui/intf/CategoryAttemptListener.kt
package cn.alvkeke.dropto.ui.intf

import cn.alvkeke.dropto.data.Category

interface CategoryAttemptListener {
    enum class Attempt {
        CREATE,
        REMOVE,
        UPDATE,
        SHOW_DETAIL,
        SHOW_CREATE,
        SHOW_EXPAND,
        DEBUG_ADD_DATA,
    }

    fun onAttempt(attempt: Attempt, category: Category?)
}

